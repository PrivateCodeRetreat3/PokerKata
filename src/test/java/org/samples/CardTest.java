package org.samples;

import org.approvaltests.Approvals;
import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;
import org.lambda.query.Query;
import org.lambda.query.Queryable;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class CardTest {
    @Test
    void testCardDeck() {
        Approvals.verifyAll("card", Deck.all() );
    }

    @Test
    void testWinners() {
        var handText = Queryable.as(
                "A♦",
                "K♦",
                "7♦ 7♣",
                "5♦ 5♣",
                "4♦ 4♣ 2♦ 2♣",
                "5♦ 5♣ 2♦ 2♣",
                "4♦ 4♣ 3♦ 3♣");
        var hands = handText.select(h -> new Hand(h)).asArray();
        CombinationApprovals.verifyAllCombinations((h11, h21) -> {
            if(h11 == h21)
            {
                throw new org.approvaltests.combinations.SkipCombination();
            }
            return Poker.whoWins(h11, h21);
        }, hands, hands);

    }

    @Test
    void testHighCardWins() {

        CombinationApprovals.verifyAllCombinations((c1, c2) -> {
            if(c1==c2)
            {
                throw new org.approvaltests.combinations.SkipCombination();
            }
            return Poker.whoWins(new Hand(c1),new Hand(c2));
        }, Deck.all(), Deck.all());

    }
}
