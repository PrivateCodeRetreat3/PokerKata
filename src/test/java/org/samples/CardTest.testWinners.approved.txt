[[A♦], [K♦]] => [A♦] 
[[A♦], [7♦ 7♣]] => [7♦ 7♣] 
[[A♦], [5♦ 5♣]] => [5♦ 5♣] 
[[A♦], [4♦ 4♣ 2♦ 2♣]] => [4♦ 4♣ 2♦ 2♣] 
[[A♦], [5♦ 5♣ 2♦ 2♣]] => [5♦ 5♣ 2♦ 2♣] 
[[A♦], [4♦ 4♣ 3♦ 3♣]] => [4♦ 4♣ 3♦ 3♣] 
[[K♦], [A♦]] => [A♦] 
[[K♦], [7♦ 7♣]] => [7♦ 7♣] 
[[K♦], [5♦ 5♣]] => [5♦ 5♣] 
[[K♦], [4♦ 4♣ 2♦ 2♣]] => [4♦ 4♣ 2♦ 2♣] 
[[K♦], [5♦ 5♣ 2♦ 2♣]] => [5♦ 5♣ 2♦ 2♣] 
[[K♦], [4♦ 4♣ 3♦ 3♣]] => [4♦ 4♣ 3♦ 3♣] 
[[7♦ 7♣], [A♦]] => [7♦ 7♣] 
[[7♦ 7♣], [K♦]] => [7♦ 7♣] 
[[7♦ 7♣], [5♦ 5♣]] => [7♦ 7♣] 
[[7♦ 7♣], [4♦ 4♣ 2♦ 2♣]] => [4♦ 4♣ 2♦ 2♣] 
[[7♦ 7♣], [5♦ 5♣ 2♦ 2♣]] => [5♦ 5♣ 2♦ 2♣] 
[[7♦ 7♣], [4♦ 4♣ 3♦ 3♣]] => [4♦ 4♣ 3♦ 3♣] 
[[5♦ 5♣], [A♦]] => [5♦ 5♣] 
[[5♦ 5♣], [K♦]] => [5♦ 5♣] 
[[5♦ 5♣], [7♦ 7♣]] => [7♦ 7♣] 
[[5♦ 5♣], [4♦ 4♣ 2♦ 2♣]] => [4♦ 4♣ 2♦ 2♣] 
[[5♦ 5♣], [5♦ 5♣ 2♦ 2♣]] => [5♦ 5♣ 2♦ 2♣] 
[[5♦ 5♣], [4♦ 4♣ 3♦ 3♣]] => [4♦ 4♣ 3♦ 3♣] 
[[4♦ 4♣ 2♦ 2♣], [A♦]] => [4♦ 4♣ 2♦ 2♣] 
[[4♦ 4♣ 2♦ 2♣], [K♦]] => [4♦ 4♣ 2♦ 2♣] 
[[4♦ 4♣ 2♦ 2♣], [7♦ 7♣]] => [4♦ 4♣ 2♦ 2♣] 
[[4♦ 4♣ 2♦ 2♣], [5♦ 5♣]] => [4♦ 4♣ 2♦ 2♣] 
[[4♦ 4♣ 2♦ 2♣], [5♦ 5♣ 2♦ 2♣]] => [5♦ 5♣ 2♦ 2♣] 
[[4♦ 4♣ 2♦ 2♣], [4♦ 4♣ 3♦ 3♣]] => [4♦ 4♣ 3♦ 3♣] 
[[5♦ 5♣ 2♦ 2♣], [A♦]] => [5♦ 5♣ 2♦ 2♣] 
[[5♦ 5♣ 2♦ 2♣], [K♦]] => [5♦ 5♣ 2♦ 2♣] 
[[5♦ 5♣ 2♦ 2♣], [7♦ 7♣]] => [5♦ 5♣ 2♦ 2♣] 
[[5♦ 5♣ 2♦ 2♣], [5♦ 5♣]] => [5♦ 5♣ 2♦ 2♣] 
[[5♦ 5♣ 2♦ 2♣], [4♦ 4♣ 2♦ 2♣]] => [5♦ 5♣ 2♦ 2♣] 
[[5♦ 5♣ 2♦ 2♣], [4♦ 4♣ 3♦ 3♣]] => [5♦ 5♣ 2♦ 2♣] 
[[4♦ 4♣ 3♦ 3♣], [A♦]] => [4♦ 4♣ 3♦ 3♣] 
[[4♦ 4♣ 3♦ 3♣], [K♦]] => [4♦ 4♣ 3♦ 3♣] 
[[4♦ 4♣ 3♦ 3♣], [7♦ 7♣]] => [4♦ 4♣ 3♦ 3♣] 
[[4♦ 4♣ 3♦ 3♣], [5♦ 5♣]] => [4♦ 4♣ 3♦ 3♣] 
[[4♦ 4♣ 3♦ 3♣], [4♦ 4♣ 2♦ 2♣]] => [4♦ 4♣ 3♦ 3♣] 
[[4♦ 4♣ 3♦ 3♣], [5♦ 5♣ 2♦ 2♣]] => [5♦ 5♣ 2♦ 2♣] 
